### PR TITLE
Update ns.bench (fix another clang bug with tuple of size one)

### PR DESCRIPTION
**Pull request by ccharly**

### CODE REVIEW

**Comment by jfalcou** [comment]
So we're down providing a silly tuple local implementation ?
Can we protect it with a minimal clang version so when it's fixed upstream we can just not use it ?

File: bench/ns.bench.hpp [hunk 1]
```diff
@@ -780,6 +780,82 @@ using make_range_t = typename make_range<N>::type;
   (void)std::initializer_list<int>{((void)f(__VA_ARGS__), 0)...}                                   \
 
 } } }
+#define NS_BENCH_AUTO_DECLTYPE(...)                                                                \
+  -> decltype(__VA_ARGS__) { return (__VA_ARGS__); }                                               \
+
+#define NS_BENCH_ASM_MARKER() __asm__ __volatile__ ("cpuid")
```
Comment (jfalcou) [inline bench/ns.bench.hpp]: This being very x86 dependent, do we need it or should we protect it with BOOST-ARCH_X86 ?

Comment (ccharly) [reply bench/ns.bench.hpp]: yup indeed. I'll make it available only on x86 yes.

**Comment by ccharly** [comment]
Yes, I can do that too.

**Comment by ccharly** [comment]
Here you are! I've pushed force though... It removes your comments on the previous commit. Sorry for that. :)

**Comment by ccharly** [comment]
It seems that travis ran out of time once more. Anyway, that PR is only about benches which are not tested by travis, so I suggest we can just merge it as is.

**Pull request merged**